flow/util/def2stream.py

```diff
@@ -20,43 +20,40 @@
 for i in main_layout.each_cell():
     print("[INFO] '{0}'".format(i.name))
 
-print("[INFO] Reading DEF ...")
 main_layout.read(in_def, layoutOptions)
 
 # Clear cells
 top_cell_index = main_layout.cell(design_name).cell_index()
 
 # remove orphan cell BUT preserve cell with VIA_
 #  - KLayout is prepending VIA_ when reading DEF that instantiates LEF's via
-print("[INFO] Clearing cells...")
 for i in main_layout.each_cell():
     if i.cell_index() != top_cell_index:
         if not i.name.startswith("VIA_") and not i.name.endswith("_DEF_FILL"):
             i.clear()
 
 # Load in the gds to merge
-print("[INFO] Merging GDS/OAS files...")
 for fil in in_files.split():
     print("\t{0}".format(fil))
     main_layout.read(fil)
 
 # Copy the top level only to a new layout
-print("[INFO] Copying toplevel cell '{0}'".format(design_name))
 top_only_layout = pya.Layout()
 top_only_layout.dbu = main_layout.dbu
 top = top_only_layout.create_cell(design_name)
 top.copy_tree(main_layout.cell(design_name))
 
-print("[INFO] Checking for missing cell from GDS/OAS...")
 missing_cell = False
-regex = None
-if "GDS_ALLOW_EMPTY" in os.environ:
-    print("[INFO] Found GDS_ALLOW_EMPTY variable.")
-    regex = os.getenv("GDS_ALLOW_EMPTY")
+allow_empty = os.environ.get("GDS_ALLOW_EMPTY", "")
+regex = re.compile(allow_empty) if allow_empty else None
+
+if allow_empty:
+    print(f"[INFO] GDS_ALLOW_EMPTY={allow_empty}")
+
 for i in top_only_layout.each_cell():
     if i.is_empty():
         missing_cell = True
-        if regex is not None and re.match(regex, i.name):
+        if regex is not None and regex.match(i.name):
```

Comment:
Driveby fix: move regex compilation out of loop, don't think it matters, I have never seen def2stream take any time.